freetextresponse/public/view.js

```diff
@@ -98,6 +93,16 @@ function FreeTextResponseView(runtime, element) {
         return html;
     }
 
+    function display_responses_if_answered(response) {
+        if (!response.user_alert && response.display_other_responses) {
+            var responseHTML = get_student_responses_html(response.other_responses);
+            if (responseHTML) {
+                responseList.html(responseHTML);
+            }
+            $element.find('.responses-box').removeClass('hidden');
```

Comment:
Should this instead be inside the `if` block?
That way it only tries to remove the class if it actually pulled back HTML?

Or actually, what happens if the AJAX lookup here (from `get_student_responses_html`) doesn't find any items. Should `get_student_responses_html` instead return the "No messages to display" instead of an empty string? In that case, we can remove the `if` check completely and just update the inner html regardless.

Reply:
By default `responseList` is loaded with `<li class="no-response">No responses to show at this time</li>` if there isn't a student response. So if the AJAX call comes back empty it'll just show that default value...

Reply:
I think that relies too much on assumptions of the underlying implementation.

If we don't have any `responseHTML` and we aren't updating the content of `responseList`,
then we shouldn't be modifying the responses box.

It's possible, however unexpected, that you could start with responses, and later get back nothing.
- a response could be deleted
- an error could occur that causes the lookup to fail and instead returns an empty list

We can't make this assumption.

An easy way to do this is to just bail immediately if there isn't anything to update:

```javascript
if (!responseHtml) {
    return;
}
responseList.html(responseHtml);
$element.find('.responses-box').removeClass('hidden');
```

But still, as pointed out, this doesn't address what happens if we _do_ have responses at one point, but then later we don't.
The easy way to handle this is to just have our `get_student_responses_html` helper return either the response html or the "no-response" html.

Reply:
We can discuss more f2f, if needed.